python/tests/test_misc_literal_conversions.py

```diff
@@ -0,0 +1,308 @@
+    assert lit.to(test_type) is None
+
+
+@pytest.mark.parametrize("lit", [iceberg.literals.of("2017-08-18").to(DateType())])
+@pytest.mark.parametrize(
+    "test_type",
+    [
+        BooleanType(),
+        IntegerType(),
+        LongType(),
+        FloatType(),
+        DoubleType(),
+        TimeType(),
+        TimestampType(),
+        TimestamptzType(),
+        DecimalType(9, 2),
+        StringType(),
+        UUIDType(),
+        FixedType(1),
+        BinaryType(),
+    ],
+)
+def test_invalid_datetime_conversions(lit, test_type):
+    assert_invalid_conversions(lit, (test_type,))
+
+
+def test_invalid_time_conversions():
+    assert_invalid_conversions(
+        iceberg.literals.of("14:21:01.919").to(TimeType()),
+        [
+            BooleanType(),
+            IntegerType(),
+            LongType(),
+            FloatType(),
+            DoubleType(),
+            DateType(),
+            TimestampType(),
+            TimestamptzType(),
+            DecimalType(9, 2),
+            StringType(),
+            UUIDType(),
+            FixedType(1),
+            BinaryType(),
+        ],
+    )
+
+
+def test_invalid_timestamp_conversions():
+    assert_invalid_conversions(
+        iceberg.literals.of("2017-08-18T14:21:01.919").to(TimestampType()),
+        [
+            BooleanType(),
+            IntegerType(),
+            LongType(),
+            FloatType(),
+            DoubleType(),
+            TimeType(),
+            DecimalType(9, 2),
+            StringType(),
+            UUIDType(),
+            FixedType(1),
+            BinaryType(),
+        ],
+    )
+
+
+def test_invalid_decimal_conversions():
+    assert_invalid_conversions(
+        iceberg.literals.of(Decimal("34.11")),
+        [
+            BooleanType(),
+            IntegerType(),
+            LongType(),
+            FloatType(),
+            DoubleType(),
+            DateType(),
+            TimeType(),
+            TimestampType(),
+            TimestamptzType(),
+            DecimalType(9, 4),
+            StringType(),
+            UUIDType(),
+            FixedType(1),
+            BinaryType(),
+        ],
+    )
+
+
+def test_invalid_string_conversions():
+    assert_invalid_conversions(
+        iceberg.literals.of("abc"),
+        [BooleanType(), IntegerType(), LongType(), FloatType(), DoubleType(), FixedType(1), BinaryType()],
+    )
+
+
+def test_invalid_uuid_conversions():
+    assert_invalid_conversions(
+        iceberg.literals.of(uuid.uuid4()),
+        [
+            BooleanType(),
+            IntegerType(),
+            LongType(),
+            FloatType(),
+            DoubleType(),
+            DateType(),
+            TimeType(),
+            TimestampType(),
+            TimestamptzType(),
+            DecimalType(9, 2),
+            StringType(),
+            FixedType(1),
+            BinaryType(),
+        ],
+    )
+
+
+def test_invalid_fixed_conversions():
+    if sys.version_info[0] >= 3:
```

Comment:
Are we supporting Python 2.7?

I know that in some respects, Python 2.7 won't die. But if we don't have to support it, that would be great.

Otherwise, does the `setup.py` file or anything else provide us a means of indicating required Python version?

Reply:
Thank you @kbendick , I think that's a really good callout. 

I did look around existing code in the new module and find nothing to limit the minimal python version we support. I think if we plan to only support it for 3 onwards (with some helpful python feature we can use as part of development), we shall do it. I think that's the topic warrant the discussion on next Python meet.

Once we decides, I can go over the code base and fix the check like this, what do you think?

Reply:
Good catch @kbendick, I think there was an agreement early on to support python 3.7+ and it's set in the [setup.cfg](https://github.com/apache/iceberg/blob/master/python/setup.cfg#L44) file. The current state of python packaging feels like it's in flux (maybe it always has been?) and it's hard to tell where the right place to set this stuff is. I **think** the current standard is to use the pyproject.toml+setup.cfg combo...

@dramaticlly for these if statements you can definitely replace them with the assumption that python 3 is being used.

Reply:
Sounds good, thank you Sam for the clarification. With that being understood, I will remove those python version checks :)